### DISCUSSION TITLE
Can't upload assets as regular user

### DISCUSSION THREAD

**Discussion by humboldtux** [requarks/wiki#3328] [Error / Bug Report]
**Description**
I can't upload assets as regular user. Regular user belongs to a group with a page rule with all possible permissions on the page i want to upload assets to. But i get error "Error during upload" when i try to upload an asset.

With a user member of the built-in Administrators group, i can upload the same file on the same page.

As administrator i can create an assert folder and it is shown in the Assets Medias dialog. As regular user, i can create a folder,
no error is shown, but the asset folder is not shown. I get an error if i try to create an asset folder previously created by administartor, even if i can't see this previously created folder.

In my Nginx logs i can see the following messages for the failed regular user upload:

```
"POST /graphql HTTP/2.0" 200 87 "https://website/page"
"POST /u HTTP/2.0" 403 95 "https://website/page"
"POST /u HTTP/2.0" 403 95 "https://website/page"
```
And these for the successful administrator upload

```
"POST /graphql HTTP/2.0" 200 218 "https://website/page"
"POST /u HTTP/2.0" 200 2 "https://website/page"
"POST /graphql HTTP/2.0" 200 221 "https://website/page"
```

**Host Info (please complete the following information):**
 - OS: Debian 10
 - Wiki.js version: Wiki.js 2.5.170
 - Database engine: postgresql 11.10

**Comment by dualmacops** [comment]
Does the group have the write:assets in permission and the Upload Assets permission in page rules?

**Comment by humboldtux** [comment]
Yes it does: all assets permissions and all assets page rules.

**Comment by lwcode** [comment]
I meet the same problem, I set users to have all possible permissions, but users can't even view images and documents inserted in pages , when click the document's link, the web display only one world "Forbidden"
I have find the solution：[this issues](https://github.com/Requarks/wiki/issues/2505#issuecomment-699669893)
#2505

**Reply by jamestch** [reply]
the solution page is not found, can you help?

**Reply by Siggix64** [reply]
Same problem here. Would be grateful for a solution

**Reply by sarah-lot** [reply]
I found a solution for me. You must define a Pagerule that allows explicit on the Asset folder. Like this.
![image](https://user-images.githubusercontent.com/46600833/197727532-c1b6770d-df30-463f-8267-39c483983880.png)

Now it is possible to upload and use assets under the path /external

**Comment by vt102** [comment]
I had this issue.  The file I was uploading was larger than the allowed default limit, 5242880 bytes.  You can change this under Administration / Security / Uploads.

**Comment by joshuaherr** [comment]
Hello,
I had the same issue after throwing up an Nginx Server in front of my Wiki JS server. The issue for me was that - though the upload size was configured large enough (as @vt102 suggested above) - the nginx server was not configured to allow POST request uploads large enough for the images I was uploading.

The solution for me was to go into the nginx.conf file and add this line (or change it if you have already defined the client-max-body-size):

`client_max_body_size 50M;`

Add that line to either the `http` block or the `server` block (obviously you can change the max upload size from 50 MB if you would like).

After you add or modify that line, don't forget to reload the nginx configurations:

(on Windows):
`nginx.exe -s reload`

**Comment by mohanapriya-8ai** [comment]
I cannot upload files even as an administrator. I've created a new group and gave it all possible access to the root directory, assets directory but I still get file upload error. My files are stored on the home directory directly. 
![image](https://github.com/user-attachments/assets/62faaf08-2667-4f53-8d01-31b35abe87ae)
